TACC/tup-ui
# Task/TUP-366: ticket detail view

## Overview:

Replicate the functionality of the TicketModal used in Core-Portal.
The modal allows a user to view a ticket's history (i.e. past communications & status changes) and to submit a reply to RT.
Reference design from CMD: https://xd.adobe.com/view/cc3dc245-2792-4e10-bdc1-50d2a7d32979-996e/screen/77ae9cd7-278b-4948-81a6-a85a891dda08/

- [TUP-366](https://jira.tacc.utexas.edu/browse/TUP-366)

## Changes:

## Testing:
1. Go to http://localhost:8000/dashboard/
2. Click on one of your past ticket's text link.
3. Test out the modal: 
- Clicking on a card header should expand the card.
- If you have a ticket with attachments an attachments table should appear, which should allow you to download them.
- Test the reply form. Successfully submitting a reply should cause the ticket history card list on the left to re-render with the newly submitted reply.
- Test the test suite in a terminal window: `npx nx test tup-components`

## UI:
![image](https://user-images.githubusercontent.com/79928051/205106751-fae2adad-07dc-48d9-bff6-f82dfff8d8a3.png)

## Notes:
The response from tup-services is slightly different than what we get from Core Portal backend. Therefore, you'll likely see more ticket history cards on the left side of the modal than you would in Core Portal.

We're also missing some icons (e.g. collapse icon for the card headers and the attachment icon). We'll need to re-integrate the necessary CSS -- I've kept the classNames and calls to the ICON component that Core Portal uses for now.
